modules/common/src/main/scala/com/snowplowanalytics/snowplow/postgres/streaming/KinesisSink.scala

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2020-2021 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.postgres.streaming
+
+import java.nio.ByteBuffer
+import java.util.UUID
+
+import scala.concurrent.ExecutionContext
+import scala.jdk.FutureConverters._
+import scala.util.{Success, Failure}
+
+import cats.effect.{Async, Resource, Blocker, Sync}
+import cats.implicits._
+
+import fs2.aws.internal.{KinesisProducerClientImpl, KinesisProducerClient}
```

Comment:
This is not a criticism, just a comment:

We probably are not supposed to use `internal` classes, and they are more likely to change api interface between versions.  However, I know this lib is not perfectly designed, so if this is the best way to get the functionality then I trust your judgement.

Reply:
This is related with [this issue](https://github.com/snowplow-incubator/snowplow-postgres-loader/pull/43#issuecomment-900869265). We decided with @chuwy to not use `Pipe` in order to solve the problem I described in the mentioned comment. However, fs2-aws libraryu was returning only `Pipe` for producer. In order to use it as effect, I had to use some internal components. But, you are right definitely. This should be last resort.